### PR TITLE
Add tests for prim types.

### PR DESCRIPTION
**Pull request by katlasik**
PR for the issue: https://github.com/zio/zio-cli/issues/36

### CODE REVIEW

File: zio-cli/shared/src/main/scala/zio/cli/files/FileSystem.scala [hunk 1]
```diff
@@ -0,0 +1,30 @@
+package zio.cli.files
+
+import zio.{ IO, UIO }
+import java.nio.file.{ Files => JFiles, Path => JPath, Paths => JPaths }
+
+trait FileSystem {
```
Comment (katlasik) [inline zio-cli/shared/src/main/scala/zio/cli/files/FileSystem.scala:6]: Added trait handling file io operations to be able to mock for tests.

**Comment by shankarshastri** [comment]
I was taking a look at #42 , And was adding tests for ones missing and not covered as part of PrimType tests. But looks like you have covered most of it.
Can you also please take care of Enumeration. So that as part of #42 , I can document Args.scala 😄 , 
```scala
val anyEnumeration: Gen[Any, (String, Int)] = Gen.fromIterable(List[(String, Int)](("Green", 0), ("Red", 1), ("Blue", 2)))

suite("Enumeration Suite") {
      testM("validate returns proper Enumeration") {
        checkM(anyEnumeration) {  enumeration =>
          assertM(PrimType.Enumeration(enumeration).validate(enumeration._1, ParserOptions.default))(equalTo(enumeration._2))
        }
    }},
```

Please let us know your thoughts @jdegoes.

**Comment by katlasik** [comment]
@shankarshastri I already added suite for enumeration:
https://github.com/zio/zio-cli/pull/66/files#diff-777906a04e5faa60b3a3c5322df67ea9386e83dd3a3067398131d4aab0714b2eR27

**Comment by shankarshastri** [comment]
@katlasik , my bad. Thanks for covering !!
And looks like you have covered most of #42 as well.
@jdegoes as part of #42, Please let me know if there is anything else pending then 😄 ?

**Review by jdegoes — APPROVED** [review]

**Comment by jdegoes** [comment]
Great work. Thanks for this contribution!

**Pull request merged**